src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1348,126 +1051,23 @@ namespace winrt::Microsoft::Terminal::Control::implementation
-                    if (!_terminal->IsSelectionActive())
-                    {
-                        _lastMouseClickPosNoSelection = cursorPosition;
-                    }
-                }
-
-                // We reset the active selection if one of the conditions apply:
-                // - shift is not held
-                // - GH#9384: the position is the same as of the first click starting the selection
-                // (we need to reset selection on double-click or triple-click, so it captures the word or the line,
-                // rather than extending the selection)
-                if (_terminal->IsSelectionActive() && (!shiftEnabled || _lastMouseClickPosNoSelection == cursorPosition))
-                {
-                    // Reset the selection
-                    _terminal->ClearSelection();
-                    _selectionNeedsToBeCopied = false; // there's no selection, so there's nothing to update
-                }
-
-                if (shiftEnabled && _terminal->IsSelectionActive())
-                {
-                    // If shift is pressed and there is a selection we extend it using the selection mode
-                    // (expand the "end"selection point)
-                    _terminal->SetSelectionEnd(terminalPosition, mode);
-                    _selectionNeedsToBeCopied = true;
-                }
-                else if (mode != ::Terminal::SelectionExpansionMode::Cell || shiftEnabled)
-                {
-                    // If we are handling a double / triple-click or shift+single click
-                    // we establish selection using the selected mode
-                    // (expand both "start" and "end" selection points)
-                    _terminal->MultiClickSelection(terminalPosition, mode);
-                    _selectionNeedsToBeCopied = true;
-                }
-
-                if (_terminal->IsSelectionActive())
-                {
-                    // GH#9787: if selection is active we don't want to track the touchdown position
-                    // so that dragging the mouse will extend the selection rather than starting the new one
-                    _singleClickTouchdownPos = std::nullopt;
-                }
-
-                _renderer->TriggerSelection();
-            }
-            else if (point.Properties().IsRightButtonPressed())
-            {
-                if (_settings.CopyOnSelect() || !_terminal->IsSelectionActive())
-                {
-                    // CopyOnSelect right click always pastes
-                    PasteTextFromClipboard();
-                }
-                else
-                {
-                    CopySelectionToClipboard(shiftEnabled, nullptr);
-                }
-            }
+            const auto contactRect = point.Properties().ContactRect();
+            auto anchor = winrt::Windows::Foundation::Point{ contactRect.X, contactRect.Y };
+            _interactivity->Touched(anchor);

```

Comment:
may want to name this `TouchDown` or something in case we ever want to change things up a bit.

Reply:
(which means, "add more event handlers")